### PR TITLE
Add steps for unit_tests and specification

### PR DESCRIPTION
**Pull request by FOLLGAD**
Generates a specification based on requirements and then self-evaluates that spec and fills in any uncertainties. Then generates unit-tests before writing the actual code.

### CODE REVIEW

File: gpt_engineer/steps.py [hunk 1]
```diff
@@ -50,13 +50,43 @@ def clarify(ai: AI, dbs: DBs):
...
+
+    dbs.memory['specification'] = messages[-1]['content']
+
+    return messages
+
+def pre_unit_tests(ai: AI, dbs: DBs):
+    '''
+    Generate unit tests based on the specification, that should work.
+    '''
+    messages = [ai.fsystem(setup_sys_prompt(dbs)), ai.fsystem(f"Main prompt: {dbs.input['main_prompt']}"), ai.fsystem(f"Specification:\n\n{dbs.memory['specification']}")]
```
Comment (AntonOsika) [inline gpt_engineer/steps.py:71]:
```suggestion
    messages = [ai.fsystem(setup_sys_prompt(dbs)), ai.fuser(f"Instructions: {dbs.input['main_prompt']}"), ai.fuser(f"Specification:\n\n{dbs.memory['specification']}")]
```

File: gpt_engineer/steps.py [hunk 2]
```diff
@@ -50,13 +50,43 @@ def clarify(ai: AI, dbs: DBs):
...
+
+
 def run_clarified(ai: AI, dbs: DBs):
     # get the messages from previous step
-    messages = json.loads(dbs.logs[clarify.__name__])
 
     messages = [
         ai.fsystem(setup_sys_prompt(dbs)),
-    ] + messages[1:]
+        ai.fsystem(f"Main prompt: {dbs.input['main_prompt']}"),
```
Comment (AntonOsika) [inline gpt_engineer/steps.py:86]:
```suggestion
        ai.fsystem(f"Instructions: {dbs.input['main_prompt']}"),
```

File: identity/respec [hunk 3]
```diff
@@ -0,0 +1,8 @@
+You are a GPT-Engineer, an AI developed to write programs. You have been asked to review a specification for a new feature.
```
Comment (AntonOsika) [inline identity/respec:1]:
```suggestion
You are a pragmatic principal engineer at Google. You have been asked to review a specification for a new feature.
```

**Comment by AntonOsika** [comment]
Fantastic

**Comment by patillacode** [comment]
With your permission @FOLLGAD 
I am removing merged branches to avoid cluttering.

**Pull request merged**